### PR TITLE
docs: readme update with the Azure DevOps Addon

### PR DESCRIPTION
**Pull request by jycouet**
Just a simple PR to add the info about the addon on Azure DevOps.

### CODE REVIEW

**Comment by rarkins** [comment]
Hi @jycouet thanks for this suggestion. A few things I'd like to do first:

1. I think it's important to differentiate that this is a "third party" implementation rather than something I'm running, i.e. it's unofficial
2. I'd like @JamieMagee to take a look if he agrees this is the best way to run Renovate on Azure DevOps. He's done a lot of work on Azure DevOps lately and is best placed to study this
3. Is this really an app, or more just like a pipeline script? It looks to me to be more like a CircleCI Orb than the other Renovate Apps. i.e. it needs your own pipelines to run, it doesn't respond to webhooks, etc.
4. How does this stay up to date with Renovate? e.g. does it just install the latest release every time, or do you have a task yourself to update it to newer Renovate versions manually?
5. If we agree on the way to go ahead, let's update the description/logo and fix broken links

**Comment by jycouet** [comment]
Thx @rarkins for the quick reply :)
1. Yes fully agree, it's a third party & unofficial
2. Yes, if a second eye can look at it I would be happy 👍. (BTW, if you need help on Azure DevOps in general, you can ping me)
3. It's between an app and a pipeline script. Lot of things are done for you to setup the pipeline without too many questions... The "Addon" takes care of the endpoint, repo, token, ...
4. Each time the pipeline is running, the addon is installing Renovate (by defaut it's taking @latest, but you can configure a dedicated version if you want)
5. I will fix the broken img now. For the description/logo, I'm up to any suggestion :)

File: readme.md [hunk 1]
```diff
@@ -35,13 +35,15 @@ Renovate was released in 2017 and already is widely used in the developer commun
 
 ## Using Renovate
 
-The easiest way to use Renovate if you are hosted on GitHub.com or GitLab.com is to install the hosted Renovate app.
+The easiest way to use Renovate if you are hosted on GitHub.com or GitLab.com or DevOps (microsoft) is to install the hosted Renovate app.
```
Comment (JamieMagee) [inline readme.md]: I'm not sure of the best way to phrase this, but I think we should differentiate between GitHub.com and GitLab.com, where Renovate is run by us; and on Azure DevOps, where renovate is self-hosted.

Maybe make it a separate sentence? Something like

> The easiest way to use Renovate if you are hosted on GitHub.com or GitLab.com is to install the hosted Renovate app. On Azure DevOps you can install Renovate as an extension from the marketplace.

File: readme.md [hunk 2]
```diff
@@ -35,13 +35,15 @@ Renovate was released in 2017 and already is widely used in the developer commun
...
 
 For GitHub, go to [https://github.com/apps/renovate](https://github.com/apps/renovate) to install
 it now. A GitHub Marketplace plan is required only for private repositories. [More details on the GitHub App installation](https://docs.renovatebot.com/install-github-app/).
 
 For GitLab, log into [app.renovatebot.com/dashboard](https://app.renovatebot.com/dashboard) for a convenient way to add Renovate Bot to your project. Currently the GitLab app is free for both public and private repositories. [More details on the GitLab App installation](https://docs.renovatebot.com/install-gitlab-app/).
 
+For DevOps (microsoft), go to the marketplace and add the [RenovateMe](https://marketplace.visualstudio.com/items?itemName=jyc.vsts-extensions-renovate-me) addon to your organization. Then create a pipeline with this new task and renovate will do his job.
```
Comment (JamieMagee) [inline readme.md]: I would write it this like:

> For Azure DevOps, visit the Visual Studio Marketplace and install the [Renovate Me](https://marketplace.visualstudio.com/items?itemName=jyc.vsts-extensions-renovate-me) extension in your organisation. From there you can create a pipeline with the `RenovateMe` task.

From the source of your extension, it looks like the build task is just called `RenovateMe`, is that right? Also, as Azure DevOps support is a little more complicated, we might want to create additional help pages.

**Comment by JamieMagee** [comment]
Hey @jycouet, it's a big coincidence that you opened this now. I was just looking at how we could integrate Renovate with Azure DevOps better, over the weekend, and I saw your extension. It looked like it had been abandoned, but I'm glad to see it's still maintained.

Your extension covers pretty much how I run Renovate on Azure DevOps. I do it all in a YAML pipeline, authenticate to private npm repo, etc. and then call `npx renovate`.

One thing I do wonder, is if it's possible to have your build task depend on another build task (To install node, yarn, npm, etc.). I can't find anywhere this has been done before, but it's worth asking if we can bundle it all in 1 task.

In the long term we'd like to authenticate to Azure DevOps using OAuth, similar to how it's done on GitHub and GitLab right now, but this is definitely a good interim solution.

Would you consider forking/moving your extension to the renovatebot organisation, and maintaining it from there? My main worry is that we might get issues here, which we can't fix. No problem if you don't want to, it's only a suggestion.

File: readme.md (hunk 1 above)
Comment (jycouet) [reply readme.md]: Yes, very good. I'll do the update.

File: readme.md (hunk 2 above)
Comment (jycouet) [reply readme.md]: As you saw, there is already a ```readme.md``` there to explain what to do with this Task in the pipeline. So maybe we can put here: 

>For Azure DevOps, visit the Visual Studio Marketplace and install the Renovate Me extension in your organisation. From there you can create a pipeline with the RenovateMe task as described in the addon description.

**Comment by jycouet** [comment]
> Hey @jycouet, it's a big coincidence that you opened this now. I was just looking at how we could integrate Renovate with Azure DevOps better, over the weekend, and I saw your extension. It looked like it had been abandoned, but I'm glad to see it's still maintained.
>
Yes, I was not really active in the last months... But let's go back to what I love: Coding 👨‍💻 

> Your extension covers pretty much how I run Renovate on Azure DevOps. I do it all in a YAML pipeline, authenticate to private npm repo, etc. and then call `npx renovate`.
> 
Good to hear!

> One thing I do wonder, is if it's possible to have your build task depend on another build task (To install node, yarn, npm, etc.). I can't find anywhere this has been done before, but it's worth asking if we can bundle it all in 1 task.
> 
Today, I explain in the readme that you should add other build tasks to make sure you have node and yarn. The 2 other build tasks are very simple and exist for this purpose. I feel that if it's inside RenovateMe it would be a mix of concern.

Like this, people can run the pipelines on host (and install everything all the time) or on local build where they know that they have all the stuff installed... it bring modularity without being too complex.
> In the long term we'd like to authenticate to Azure DevOps using OAuth, similar to how it's done on GitHub and GitLab right now, but this is definitely a good interim solution.
> 
I'm not sure to follow? This is the point of the addon! It's taking take of the authentication to the Repo. So why do you want to manually authenticate to Azure DevOps using OAuth?

> Would you consider forking/moving your extension to the renovatebot organisation, and maintaining it from there? My main worry is that we might get issues here, which we can't fix. No problem if you don't want to, it's only a suggestion.
>
I'm open... Why not... Maybe it's not a top priority? And I have 2 other small addon on the same repo, so I would have to change few things before. Let's see how we can progress first?

**Comment by ViceIce** [comment]
@jycouet @JamieMagee Do you know that we have a slim docker image [`renovate/renovate:slim`](https://hub.docker.com/r/renovate/renovate) now?

With https://github.com/renovatebot/renovate/pull/4387 integrated you can simply user docker from the pipeline and don't have to think about third party deps.

**Comment by jycouet** [comment]
Thx @vicelce, I didn't know. It's good for people hosting their own build machine. Still Renovate Me will help configuring the task itself 👍.

File: readme.md (hunk 2 above)
Comment (JamieMagee) [reply readme.md]: Yeah, sounds good.

One thing though, you might want to provide a YAML build sample. I think most new builds are created this way, as opposed to using the older GUI builds.

**Comment by JamieMagee** [comment]
Sorry for the late response

>I'm not sure to follow? This is the point of the addon! It's taking take of the authentication to the Repo. So why do you want to manually authenticate to Azure DevOps using OAuth?

I wasn't talking about RenovateMe, but how we want to run Renovate on Azure DevOps in the future. RenovateMe is self-hosted, but we want to offer Renovate as a hosted service, similar to how it is an app on GitHub.

>I'm open... Why not... Maybe it's not a top priority? And I have 2 other small addon on the same repo, so I would have to change few things before. Let's see how we can progress first?

Good idea. Let's get these documentation changes merged first, and then discuss this later.

**Comment by jycouet** [comment]
> I wasn't talking about RenovateMe, but how we want to run Renovate on Azure DevOps in the future. RenovateMe is self-hosted, but we want to offer Renovate as a hosted service, similar to how it is an app on GitHub.
> 

I don't know how to do this, but would be happy to contribute 👍 
Let me know when you want to start this activity.

Yeah, let's validate the PR for now :)

**Pull request merged**